src/popup/components/form/FormInputWithQr.vue

```diff
@@ -0,0 +1,78 @@
+<template>
+  <FormTextarea
+    class="form-input-with-qr"
+    :model-value="modelValue"
+    :label="label"
+    :placeholder="placeholder"
+    size="xxs"
+    resizable
+    @input="$emit('update:modelValue', $event.target.value)"
+  >
+    <template #label-after>
+      <BtnPlain
+        class="scan-button"
+        data-cy="scan-button"
+        @click="scanQr()"
+      >
+        <QrScanIcon />
+      </BtnPlain>
+    </template>
+  </FormTextarea>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue';
+import { MODAL_READ_QR_CODE } from '@/constants';
+import { useModals } from '@/composables';
+
+import FormTextarea from './FormTextarea.vue';
+import BtnPlain from '../buttons/BtnPlain.vue';
+
+import QrScanIcon from '../../../icons/qr-scan.svg?vue-component';
+
+export default defineComponent({
+  components: {
+    FormTextarea,
+    BtnPlain,
+    QrScanIcon,
+  },
+  props: {
+    modelValue: { type: String, default: '' },
+    placeholder: { type: String, default: '' },
+    label: { type: String, default: '' },
+    qrIcon: { type: String, default: '' },
```

Comment:
Probably this can be removed as the scan qr modal does not support providing custom icons.